### PR TITLE
rosetta

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: rosetta-mainnet/.env [hunk 1]
```diff
@@ -0,0 +1,22 @@
+SHARD_0=0
```
Comment (andreibancioiu) [inline rosetta-mainnet/.env:1]: Well, a bit over-engineered constants. Can remove them and use the numbers / string where needed.

Oh, I see the same `.env` for the classic Observing Squad as well. Can be kept, for consistency, understood.

File: rosetta-mainnet/elrond-node-obs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+FROM golang:1.14.9 as builder
```
Comment (andreibancioiu) [inline rosetta-mainnet/elrond-node-obs:1]: Perhaps also here we can use the `folder/Dockerfile` approach. @AdoAdoAdo?

File: rosetta-mainnet/elrond-proxy [hunk 3]
```diff
@@ -0,0 +1,23 @@
+FROM golang:1.14.9 as builder
```
Comment (andreibancioiu) [inline rosetta-mainnet/elrond-proxy:1]: Perhaps also here we can use the `folder/Dockerfile` approach. @AdoAdoAdo? And for the testnet as well.

File: rosetta-testnet/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+     - IP=10.0.0.2
+   networks:
+     elrond-squad:
+       ipv4_address: 10.0.0.2
+
+networks:
+  elrond-squad:
+    ipam:
+      config:
+        - subnet: 10.0.0.0/24
```
Comment (andreibancioiu) [inline rosetta-testnet/docker-compose.yml:81]: I wonder if we should add the subnet as an env variable. @AdoAdoAdo, @valentin-lup? But not now, in the future.

**Review by andreibancioiu — APPROVED** [review]

File: rosetta-testnet/start_stack.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+STACK_FOLDER=~/MyObservingSquadTestnet
+KEYS_FOLDER=${STACK_FOLDER}/keys
+
+#Create the folder structure for the observer stack
+mkdir -p ${STACK_FOLDER}/{proxy,node-0,node-1,node-2,node-metachain}/{config,logs}
+mkdir -p ${STACK_FOLDER}/{node-0,node-1,node-2,node-metachain}/db
+mkdir -p ${KEYS_FOLDER}
+
+#Generate Keys and place them in their respective folders
```
Comment (AdoAdoAdo) [inline rosetta-testnet/start_stack.sh:19]: do we generate new keys every time?

**Review by AdoAdoAdo — CHANGES_REQUESTED** [review]

File: rosetta-mainnet/.env [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+OBSERVER_DIR_2=~/MyObservingSquad/node-2
+OBSERVER_DIR_M=~/MyObservingSquad/node-metachain
+
+IP_0=10.0.0.6
+IP_1=10.0.0.5
+IP_2=10.0.0.4
+IP_M=10.0.0.3
+
+NODE_TAG=elrond-rosetta-observer-mainnet:e1.1.0.1
+PROXY_TAG=elrond-rosetta-proxy-mainnet:e1.1.0.1
```
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-mainnet/.env:22]: done.

File: rosetta-mainnet/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+     - IP=10.0.0.2
+   networks:
+     elrond-squad:
+       ipv4_address: 10.0.0.2
+
+networks:
+  elrond-squad:
+    ipam:
+      config:
+        - subnet: 10.0.0.0/24
```
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-mainnet/docker-compose.yml:81]: done.

File: rosetta-mainnet/elrond-proxy (hunk 3 above)
Comment (AdoAdoAdo) [reply rosetta-mainnet/elrond-proxy:1]: I also suggested to go for folder/Dockerfile for both

File: rosetta-mainnet/elrond-node-obs (hunk 2 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-mainnet/elrond-node-obs:1]: done.

File: rosetta-mainnet/elrond-proxy (hunk 3 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-mainnet/elrond-proxy:1]: done.

File: rosetta-testnet/.env [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+OBSERVER_DIR_2=~/MyObservingSquad/node-2
+OBSERVER_DIR_M=~/MyObservingSquad/node-metachain
+
+IP_0=10.0.0.6
+IP_1=10.0.0.5
+IP_2=10.0.0.4
+IP_M=10.0.0.3
+
+NODE_TAG=elrond-rosetta-observer-testnet:v1.1.3
+PROXY_TAG=elrond-rosetta-proxy-testnet:v1.1.3
```
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-testnet/.env:22]: done.

File: rosetta-testnet/docker-compose.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-testnet/docker-compose.yml:81]: done.

File: rosetta-testnet/elrond-node-obs [hunk 9]
```diff
@@ -0,0 +1,35 @@
+FROM golang:1.14.9 as builder
```
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-testnet/elrond-node-obs:1]: done.

File: rosetta-testnet/elrond-proxy [hunk 10]
```diff
@@ -0,0 +1,23 @@
+FROM golang:1.14.9 as builder
```
[the remark this answers is not in the document]
Comment (miiu96) [reply rosetta-testnet/elrond-proxy:1]: done.

File: rosetta-mainnet/elrond-proxy (hunk 3 above)
Comment (miiu96) [reply rosetta-mainnet/elrond-proxy:1]: done.

File: rosetta-testnet/start_stack.sh (hunk 5 above)
Comment (miiu96) [reply rosetta-testnet/start_stack.sh:19]: I'm not sure. Maybe @andreibancioiu has the answer ?

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**